### PR TITLE
chore(cli): deprecate `--configuration` option in `run-macos` 

### PR DESCRIPTION
**Pull request by szymonrybczak**
#### Please select one of the following
- [ ] I am removing an existing difference between facebook/react-native and microsoft/react-native-macos :thumbsup:
- [ ] I am cherry-picking a change from Facebook's react-native into microsoft/react-native-macos :thumbsup:
- [x] I am making a fix / change for the macOS implementation of react-native
- [ ] I am making a change required for Microsoft usage of react-native

## Summary:

In future we would like to replace custom implementation of `run-macos` with the one that is inside created [`cli-platform-app`](https://github.com/react-native-community/cli/tree/main/packages/cli-platform-apple) package inside https://github.com/react-native-community/cli. To unify implementations between all Apple's OOT platforms.  To avoid creating _many_ breaking changes I'm deprecating old options that are under different name in new implementation. 

## Changelog:

[GENERAL] [DEPRECATED] - Deprecate `--configuration` option from `run-macos` command.

## Test Plan:

1. Setup CLI locally.
2. Run `run-macos --configuration` -> you should see the warning.
3. Run `run-macos --project-path` -> you should also see the warning.

### CODE REVIEW

File: packages/react-native/local-cli/runMacOS/runMacOS.js [hunk 1]
```diff
@@ -47,6 +48,13 @@ function runMacOS(_, _ctx, args) {
     );
   }
 
+  if (args.configuration) {
+    logger.warn(
+      'Argument --configuration has been deprecated and will be removed in a future release, please use --mode instead.',
+    );
+    args.mode = args.configuration;
```
Comment (tido64) [inline packages/react-native/local-cli/runMacOS/runMacOS.js:55]: If `mode` is already set, wouldn't this override it?

File: packages/react-native/local-cli/runMacOS/runMacOS.js [hunk 2]
```diff
@@ -278,9 +282,15 @@ module.exports = {
...
       default: 'Debug',
     },
+    {
+      name: '--mode [string]',
+      description:
+        'Explicitly set the scheme configuration to use. This option is case sensitive.',
+    },
     {
       name: '--scheme [string]',
       description: 'Explicitly set Xcode scheme to use',
```
Comment (tido64) [inline packages/react-native/local-cli/runMacOS/runMacOS.js:296]: Based on the description, how does `--mode` differ from `--scheme`?

File: packages/react-native/local-cli/runMacOS/runMacOS.js (hunk 1 above)
Comment (szymonrybczak) [reply packages/react-native/local-cli/runMacOS/runMacOS.js:55]: Yeah, true. Fixed.

File: packages/react-native/local-cli/runMacOS/runMacOS.js (hunk 2 above)
Comment (szymonrybczak) [reply packages/react-native/local-cli/runMacOS/runMacOS.js]: Changed, should be more clear now.

**Review by Saadnajmi — APPROVED** [review]

**Pull request merged**